IonDotnet/Internals/Text/UnicodeStream.cs

```diff
@@ -20,9 +21,9 @@ public UnicodeStream(Stream inputStream, Encoding encoding)
         {
             if (!inputStream.CanRead)
                 throw new ArgumentException("Input stream must be readable", nameof(inputStream));
-
             _streamReader = new StreamReader(inputStream, encoding);
             _unreadStack = new Stack<int>();
+            remainingChars = inputStream.Length;
```

Comment:
It seems like not all implementations of `Stream` will support `Length`: https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?view=netframework-4.8#System_IO_Stream_Length

Is there a way to implement this fix without using `Stream.Length`?

Reply:
Update the pull request as we discussed.
If an implementation does not support `Length` property, there won't be any check for short sequences.

Created issue #101 to keep track of this.